src/main/java/org/damap/base/rest/invenio_damap/InvenioDAMAPResource.java

```diff
@@ -0,0 +1,171 @@
+  DmpService dmpService;
+  InvenioDAMAPService invenioDAMAPService;
+
+  @ConfigProperty(name = "invenio.shared-secret")
+  String sharedSecret;
+
+  @Inject JWTParser parser;
+
+  /*
+   * Maybe make it configurable?
+   * Could be more elaborate but should suffice for PoC
+   * Could also go into the {@link org.damap.base.security.SecurityService},
+   * but keeping everything together for PoC
+   *
+   */
+  private JsonWebToken validateAuthHeader(HttpHeaders headers) {
+    String jwt_token = headers.getHeaderString("X-Auth");
+
+    if (jwt_token != null && !jwt_token.isEmpty()) {
+      try {
+        JsonWebToken jwt = parser.verify(jwt_token, sharedSecret);
+
+        long exp = jwt.getExpirationTime();
+        long currentTime = System.currentTimeMillis() / 1000;
+
+        if (currentTime >= exp) throw new UnauthorizedException("Token expired.");
+
+        return jwt;
+      } catch (ParseException e) {
+        return null;
+      }
+    }
+    return null; // No Authorization header or not in the correct format
+  }
+
+  // Checks provided jwt.
+  private JsonWebToken checkIfUserIsAuthorized(HttpHeaders headers) {
+
+    JsonWebToken jwt = validateAuthHeader(headers);
+    boolean authorized = jwt != null;
+
+    if (!authorized) {
+      throw new UnauthorizedException("User unauthorized.");
+    }
+    return jwt;
+  }
+
+  // Resolves user identity based on the DMPs and returns dmp list and list of
+  // identifiers if successful.
+  private SimpleEntry<List<DmpListItemDO>, List<String>> resolveDmpsAndIds(JsonObject identifiers) {
+    List<DmpListItemDO> personDmpList = null;
+    List<String> matchingIdentifiers = new ArrayList<>();
+
+    for (String key : identifiers.keySet()) {
+      String identifier = identifiers.getString(key);
+      List<DmpListItemDO> dmps = dmpService.getDmpListByPersonId(identifier);
+
+      // Check if returned DMPs is null or the list is empty.
+      // If yes, no resolving can take place, move to the next identifier.
+      if (dmps == null || dmps.isEmpty()) continue;
+
+      // Here dmps would contain at least one DMP so keep track of the current
+      // iterated identifier.
+      matchingIdentifiers.add(identifier);
+
+      // Keep track of the 1st list that matches at least one DMP.
+      // Then compare it with other non-null or non-empty lists.
+      if (personDmpList == null) personDmpList = dmps;
+
+      // If a DMP list is different than the previous one, a different identity was
+      // resolved.
+      else if (!personDmpList.equals(dmps))
+        throw new UnauthorizedException("Mismatch in resolved identities.");
+    }
+
+    // Check if DMP list is null. This means that no resolving was possible.
+    if (personDmpList == null) throw new NotFoundException("No valid identities were found.");
+
+    return new SimpleEntry<>(personDmpList, matchingIdentifiers);
+  }
+
+  @GET
+  public List<DmpListItemDO> getDmpListByPerson(@Context HttpHeaders headers) {
```

Comment:
We should return a list of DMPs that conforms to the maDMP standard of the RDA. So we have to map the results from here to maDMP (there should be a mapper available).

Reply:
I need a little more context here, a list of DMPDOs or the actual maDMPs? In any case, something is indeed missing. Having finalized the InvenioRDM frontend changes, the "already linked" label is not displayed and the cause seems to be the absence of the dataset in the response.

Reply:
A list of maDMPs. Some sort of pagination would also be great but we did not settle on this.
I think we had maDMPs before as well which is what the InvenioRDM implementation is based on

Reply:
Sorry about the confusion and extra work. maDMPs do not have a local identifier (i.e. database id) which is why we can not simply use a list of maDMPs. The Invenio frontend would not know which id to send in order to add a dataset to a DMP. Using a List of `DmpDO` solves this issue, and also includes the datasets (needed for some UX on Invenio side) as compared to the `DmpListItemDO`.